core/src/main/java/org/jdbi/v3/tweak/CollectorFactory.java

```diff
@@ -38,5 +37,5 @@
      * @param type the actual type of the container
      * @return the {@link Collector}
      */
-    Collector<T, ?, R> newCollector(Class<R> type);
+    Collector<?, ?, ?> newCollector(ResolvedType type);
```

Comment:
It seems like a step backward to lose the guarantee that the collector actually collects `T` into `R`.